lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_po_file_from_metadata.rb

```diff
@@ -0,0 +1,136 @@
+require_relative '../../helper/po_extended'
+require 'fastlane/action'
+require 'gettext/po'
+
+module Fastlane
+  module Actions
+    class AndroidGeneratePoFileFromMetadataAction < Action
+
+      # SPECIAL_KEYS are keys that need to be treated specially
+      SPECIAL_KEYS = %w[release_notes release_notes_previous].freeze
+      REQUIRED_KEYS_TO_COMMENT_HASH = {
+        full_description: 'Multi-paragraph text used to display in the Play Store. Limit to 4000 characters including spaces and commas!',
+        title: 'Title to be displayed in the Play Store. Limit to 30 characters including spaces and commas!',
+        short_description: 'Short description of the app to be displayed in the Play Store. Limit to 80 characters including spaces and commas!',
+        release_notes_short: nil,
+        release_notes: 'Release notes for this version to be displayed in the Play Store. Limit to 500 characters including spaces and commas!',
+        release_notes_previous: nil
+      }.freeze
+
+      REQUIRED_KEYS = REQUIRED_KEYS_TO_COMMENT_HASH.keys.map(&:to_s).freeze
+
+      OPTIONAL_KEYS_TO_LABELS_HASH = {
+        # optional_key_label: 'Description for the first app store image',
+      }.freeze
+      KEYS_TO_COMMENT_HASH = REQUIRED_KEYS_TO_COMMENT_HASH.merge(OPTIONAL_KEYS_TO_LABELS_HASH).freeze
+
+
+      def self.required_keys
+        REQUIRED_KEYS
+      end
+
+      def self.run(params)
+        @metadata_directory = params[:metadata_directory]
+        @release_version = params[:release_version]
+        @other_sources = params[:other_sources]
+
+        prefix = 'play_store'
+        all_keys = Dir[File.join(@metadata_directory, '*.txt')]
+
+        # Remove from all_keys the special keys as they need to be treated specially
+        standard_files = []
+        all_keys.each do |key|
+          standard_files.append(key) unless SPECIAL_KEYS.include? File.basename(key, '.txt')
+        end
+        # Let the helper handle standard files
+        po = PoExtended.new(:msgctxt)
+        po = Fastlane::Helper::GeneratePoFileMetadataHelper.add_header_to_po(po)
+        po = Fastlane::Helper::GeneratePoFileMetadataHelper.add_standard_files_to_po(prefix, files: standard_files, keys_to_comment_hash: KEYS_TO_COMMENT_HASH, po_obj: po)
```

Comment:
> And if you end up going with my suggestion of making the Helper instanciable and have the PoExtended be an instance variable, you could then also pass the keys_to_comment_hash at instanciation time (i.e. po = Fastlane::Helper::GeneratePoFileMetadataHelper.new(keys_to_comment_hash: KEYS_TO_COMMENT_HASH) instead of having to pass it explicitly at each call.

For this to happen I assume I need a `class` right? Or in ruby it's possibile to _init_ a Module?